src/Symfony/Component/Validator/Util/LegacyTranslatorProxy.php

```diff
@@ -22,15 +22,26 @@ class LegacyTranslatorProxy implements LegacyTranslatorInterface, TranslatorInte
 {
     private $translator;
 
-    public function __construct(TranslatorInterface $translator)
+    /**
+     * @param LegacyTranslatorInterface|TranslatorInterface $translator
+     */
+    public function __construct($translator)
     {
+        if (!$translator instanceof LegacyTranslatorInterface && !$translator instanceof TranslatorInterface) {
+            throw new \InvalidArgumentException(sprintf('The translator passed to "%s()" must implement "%s" or "%s".', __METHOD__, TranslatorInterface::class, LegacyTranslatorInterface::class));
+        }
+
         if (!$translator instanceof LocaleAwareInterface) {
```

Comment:
it should implement either `LocaleAwareInterface` or `LegacyTranslatorInterface` though (as LegacyTranslatorInterface also has `setLocale`)

Reply:
updated